jaq-core/src/lib.rs

```diff
@@ -182,20 +190,24 @@ where
 }
 
 fn to_sh(v: &Val) -> Result<String, Error> {
-    let fail = || format!("cannot escape for shell: {v}");
     Ok(match v {
         Val::Str(s) => format!("'{}'", s.replace('\'', r"'\''")),
-        Val::Arr(_) | Val::Obj(_) => return Err(Error::str(fail())),
+        Val::Arr(_) | Val::Obj(_) => {
+            return Err(Error::str(&format_args!("cannot escape for shell: {v}")))
+        }
```

Comment:
I prefer the `fail()` method, because it is shorter (line-wise).

Reply:
`Arguments`, which is returned by `format_args`, can't be returned by a closure because it is not an owned object.

I tried to improve it with an intermediate variable in https://github.com/01mf02/jaq/pull/124/commits/69f3e9d00d2d50bf2a8e35236b603f09b2369f00